library/core/src/intrinsics/simd.rs

```diff
@@ -243,12 +243,13 @@ extern "platform-intrinsic" {
     ///
     /// `T` must be a vector.
     ///
-    /// `U` must be a vector of pointers to the element type of `T`, with the same length as `T`.
+    /// `U` must be a pointer to the element type of `T`
```

Comment:
Why not make the argument of type `*const U` then? It's not necessary to make it so generic.

Same for the store operation.

Reply:
I think this is a good idea. I'll explore this

Reply:
So this is left to a future PR?

Reply:
Yeah, I think we'll need to change how we validate these monomorphizations and there will be an opportunity when we move these checks out of the codegen crate.